Python/pystate.c

```diff
@@ -900,13 +897,6 @@ PyThreadState_Clear(PyThreadState *tstate)
     if (tstate->on_delete != NULL) {
         tstate->on_delete(tstate->on_delete_data);
```

Comment:
Should this also be moved to `tstate_delete_common`, or is it unrelated to deleting the data stack?

Reply:
I have no idea what it does, so I'm not touching it :slightly_smiling_face:

Reply:
Me neither :) I tried moving it, so it seems to work fine either place. BTW, I found a clue in Include:
https://github.com/python/cpython/blob/a9e43615c2e1fc5dd60063c1509e8b1c5daad095/Include/cpython/pystate.h#L134-L143

Reply:
Also: https://docs.python.org/3/c-api/init.html#c.PyThreadState_Clear:

_"Changed in version 3.9: This function now calls the PyThreadState.on_delete callback. Previously, that happened in PyThreadState_Delete()."_

Reply:
... and here:

https://github.com/python/cpython/blob/a9e43615c2e1fc5dd60063c1509e8b1c5daad095/Modules/_threadmodule.c#L1300-L1329

Reply:
It's added by @vstinner in GH-18296 (issue 39511). I guess he knows if it should stay or follow, then :)